lib/new_relic/agent/instrumentation/net_http/instrumentation.rb

```diff
@@ -46,6 +46,10 @@ def request_with_tracing(request)
           end
         end
 
+        def openai
+          @openai ||= %i[prepend chain].include?(NewRelic::Agent.config[:'instrumentation.ruby_openai']) && NewRelic::Agent.config[:'ai_monitoring.enabled']
+        end
```

Comment:
Similar to how https://github.com/newrelic/newrelic-ruby-agent/pull/2511 delivered a new `NewRelic::Agent::LLM` base class with logic for the customer attribute aggregator to call, I'm wondering if the OpenAI library specific methods and constant should similarly be moved out of the Net::HTTP class and into their own OpenAI class.

Then the Net::HTTP class would be left with just a lone line like so:

```diff
- OPENAI_SEGMENT_PATTERN = %r{Llm/.*/OpenAI/.*}.freeze

- openai && openai_parent?(segment)
+ NewRelic::Agent::OpenAI.parent_is_openai?(segment)

- def openai; end

- def openai_parent?(segment); end
```

to reference the following:

```ruby
class NewRelic::Agent::OpenAI
  SEGMENT_PATTERN = %r{Llm/.*/OpenAI/.*}.freeze

  def self.parent_is_openai?(segment)
    return unless involved?

    segment&.parent&.name&.match?(SEGMENT_PATTERN)
  end

  def self.involved?
    @involved ||= %i[prepend chain].include?(NewRelic::Agent.config[:'instrumentation.ruby_openai']) && NewRelic::Agent.config[:'ai_monitoring.enabled']
  end
end
```

Reply:
@fallwith this is a great call! Update https://github.com/newrelic/newrelic-ruby-agent/pull/2512/commits/23ee632f744edba1b12d273bc3c6ee958271296f